### PR TITLE
Fix Chocolatey bootstrap tests

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Fixes failing bootstrap tests. It actually ended up being an issue with the file module on Windows not removing directories.

### What issues does this PR fix or reference?
Fixes: 3007.x pipeline

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

File: salt/modules/win_file.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 import stat
 import sys
 import tempfile
+from pathlib import Path
```
Comment (s0undt3ch) [inline salt/modules/win_file.py:18]:
```suggestion
import pathlib
```

File: salt/modules/win_file.py [hunk 2]
```diff
@@ -1345,41 +1346,37 @@ def remove(path, force=False):
     # Symlinks. The shutil.rmtree function will remove the contents of
     # the Symlink source in windows.
 
-    path = os.path.expanduser(path)
+    path = Path(os.path.expanduser(path))
```
Comment (s0undt3ch) [inline salt/modules/win_file.py:1349]:
```suggestion
    path = pathlib.Path(os.path.expanduser(path))
```

File: salt/modules/win_file.py [hunk 3]
```diff
@@ -1345,41 +1346,37 @@ def remove(path, force=False):
...
 
-    if not os.path.isabs(path):
-        raise SaltInvocationError(f"File path must be absolute: {path}")
+    if not path.is_absolute():
+        raise SaltInvocationError(f"File path must be absolute: {str(path)}")
```
Comment (s0undt3ch) [inline salt/modules/win_file.py:1352]:
```suggestion
        raise SaltInvocationError(f"File path must be absolute: {path}")
```

File: salt/modules/win_file.py [hunk 4]
```diff
@@ -1345,41 +1346,37 @@ def remove(path, force=False):
...
 
     # Does the file/folder exists
-    if not os.path.exists(path) and not is_link(path):
-        raise CommandExecutionError(f"Path not found: {path}")
+    if not path.exists() and not path.is_symlink():
+        raise CommandExecutionError(f"Path not found: {str(path)}")
```
Comment (s0undt3ch) [inline salt/modules/win_file.py:1356]:
```suggestion
        raise CommandExecutionError(f"Path not found: {path}")
```

File: salt/modules/win_file.py [hunk 5]
```diff
@@ -1345,41 +1346,37 @@ def remove(path, force=False):
...
             # rmdir will work now because the directory is empty
-            os.rmdir(path)
+            path.rmdir()
     except OSError as exc:
         if force:
             # Reset attributes to the original if delete fails.
-            win32api.SetFileAttributes(path, file_attributes)
-        raise CommandExecutionError(f"Could not remove '{path}': {exc}")
+            win32api.SetFileAttributes(str(path), file_attributes)
+        raise CommandExecutionError(f"Could not remove '{str(path)}': {exc}")
```
Comment (s0undt3ch) [inline salt/modules/win_file.py:1379]:
```suggestion
        raise CommandExecutionError(f"Could not remove '{path}': {exc}")
```

**Review by s0undt3ch — DISMISSED** [review]
Let's use the top level import and no need to cast to string on f-strings, that would be double casting to string.

File: salt/modules/win_file.py (hunk 1 above)
Comment (twangboy) [reply salt/modules/win_file.py]: Done

File: salt/modules/win_file.py (hunk 3 above)
Comment (twangboy) [reply salt/modules/win_file.py]: Done

File: salt/modules/win_file.py (hunk 5 above)
Comment (twangboy) [reply salt/modules/win_file.py]: Done

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**